### ISSUE TITLE
mapApp proposal: default should show all 3 argo types

### ISSUE THREAD

**Issue by dankelley** [ArgoCanada/argoFloats#478]
I mentioned this separately, but I'm making an issue to keep track of the idea.  To me, it makes sense, and therefore I plan to do it this afternoon, unless @j-harbin objects.

**Comment by j-harbin** [comment]
Ok,

I remember why we decided to have core, bgc, and deep not show up as default. In mapApp, line 311, we have `ignoreNULL=FALSE`.

When we remove this, core, bgc, and deep DO show up as default, however, the user then does not have the option to click off NO bgc, deep, or core. I can't imagine a time where they would want to do that, but it is not intuitive when you click off all of the options and floats still show. I do remember we spent a lot of time trying to figure this out, and the only answer we could come up with was to do this `ignoreNULL=FALSE` (ie. having nothing displayed by default).

Thoughts @dankelley ?

**Comment by j-harbin** [comment]
PS - See https://github.com/ArgoCanada/argoFloats/issues/467 for the issue on having nothing selected.

**Comment by dankelley** [comment]
Oh, this is ringing some bells now.  Sometimes a person might want to use this to get just a map, with no floats.  But, really, we could add that as a separate option *if we wanted*, perhaps by having a separate button called "Show only bathymetry" or whatever...

But do we want to do that?  I think not. There are other ways for folks to get maps, and it's not our responsibility to give them a nice tool here. For example, maybe they want a projection.  So we'd need a pulldown menu with 100 choices for that.  And then each projection has parameters, like the central location.  Another interface item has to be created for that.  And then what about the colour scheme? More UI elements needed for that.  So, I argue, let's not let `mapApp()` morph into a tool for making maps.  We could make another tool for that.  And, then people might want to draw stuff on top, like where they have moorings, where a ship went, etc.  How should they do this work?  They should just *write code* to do it!  That whole range of possibilities is available by writing some code (using oce or other tools).

Long story short, I vote that we go back to the thing where you had to show at least one argo type.  After all, the app is in a package about argo!

**Comment by j-harbin** [comment]
Amazing! I agree with all of the above. This is fixed in commit 216807fa5d6fd578327a1fb556bf1823e0aca22e of develop. I will now close this issue.

**Comment by dankelley** [comment]
Hm.  The interface lets me unclick all 3 types, but it still shows the last type that I had clicked.  That will be counter-intuitive to users, so I'm reopening this, in hopes that @j-harbin can fix this.

**Comment by dankelley** [comment]
During my run, I remembered why this occurs.  I think we ought to go back one step, to when the initial view showed no floats.  That way, the GUI and the graph will agree.

**Comment by j-harbin** [comment]
Done in commit 72f068ac5b43521d79c35b80c99b006842d04a01 of the develop branch.

Just to confirm, this is now the initial view of mapApp(). If you're satisfied with this feel free to close the issue :)

![Screen Shot 2021-08-13 at 12 46 09 PM](https://user-images.githubusercontent.com/58750538/129385177-d8609e93-792e-4ee9-b564-a4caa4eb8e9b.png)

**Comment by dankelley** [comment]
This looks great.  Thanks, J!!
